forge/forge/components/README.md

```diff
@@ -30,6 +30,35 @@ class MyAgent(BaseAgent):
         self.some_component = SomeComponent(self.hello_component)
 ```
 
+## Component configuration
+
+Each component can have its own configuration defined using a regular pydantic `BaseModel`.
+To make the configuration is loaded from the configuration file correctly, the component must inherit from `ConfigurableComponent[T]` where `T` is the configuration model it uses.
+`ConfigurableComponent` provides a `config` attribute that holds the configuration instance.
+It's possible to either set the `config` attribute directly or pass the configuration instance to the component's constructor.
+To see the configuration of built-in components visit [Built-in Components](./built-in-components.md).
+
+```py
+from pydantic import BaseModel
+from forge.agent.components import ConfigurableComponent
+
+class MyConfig(BaseModel):
+    some_value: str
```

Comment:
Can we force some kind of Field here so we can autogen the docs? is that too much

Reply:
If you literally mean force, then yes I can raise an exception if any field doesn't inherit from `Field` but do we really need this?

Reply:
Yeah fair, I was thinking a description etc could be required but that’s a later problem imo

Reply:
You could enforce that every field has a `description` attribute by checking `MyConfig.__fields__`